tools/autograd/load_derivatives.py

```diff
@@ -374,7 +412,8 @@ def set_up_derivatives(f: NativeFunction) -> Tuple[
         # Test to see if the use of 'grads' makes sense.
         check_grad_usage(defn_name, derivatives)
 
-        return derivatives, forward_derivatives, args_with_derivatives, non_differentiable_arg_names
+        return (derivatives, forward_derivatives, args_with_derivatives,
```

Comment:
Why do you need to return this?

Reply:
In gen_autograd_functions.py we need to associate each name with its index so that we can add each `const auto grad_name = grads[index];` alias.

Per our earlier discussion in the PR before ghstack (#63040) we decided not to have a map of each used name and its index in each derivative. But we still need that mapping in the codegen so we return it.